### PR TITLE
types/Sharedb - add an optional agent parameter to the client connection

### PR DESCRIPTION
**Pull request by eveshum**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x ] Provide a URL to documentation or source code which provides context for the suggested changes:

https://github.com/share/sharedb/blob/v1.1.0/lib/client/connection.js#L58-L61
https://github.com/share/sharedb/blob/v1.1.0/lib/backend.js#L112-L115

- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by eveshum** [comment]
@ericyhwang

File: types/sharedb/index.d.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@
 // Project: https://github.com/share/sharedb
 // Definitions by: Steve Oney <https://github.com/soney>
 //                 Eric Hwang <https://github.com/ericyhwang>
+//                 Eve Shum <https://github.com/eveshum>
```
Comment (ericyhwang) [inline types/sharedb/index.d.ts]: By putting your name and username here, the bot will start tagging you on all PRs for the sharedb types.

Do you feel comfortable enough with the sharedb type definitions to review and approve PRs on them?

File: types/sharedb/lib/client.d.ts [hunk 2]
```diff
@@ -1,9 +1,11 @@
 /// <reference path="sharedb.d.ts" />
 import * as WS from 'ws';
 import * as ShareDB from './sharedb';
+import Agent = require('./agent');
 
 export class Connection {
     constructor(ws: WebSocket | WS);
+    agent?: Agent;
```
Comment (ericyhwang) [inline types/sharedb/lib/client.d.ts]: The `agent` property [is always set](https://github.com/share/sharedb/blob/v1.1.0/lib/client/connection.js#L58-L61), with a default value of `null`. For accuracy, I'd suggest:

```suggestion
    agent: Agent | null;
```

It'd also be good to add a JSDoc comment indicating that only same-process server connections will have an agent set.

**Review by ericyhwang — COMMENTED** [review]
The PR template asks for links to source code to support the change to the types. It's helpful for when the types reviewer isn't closely familiar with the JS source code.

Can you add these to the relevant part of the PR description?

https://github.com/share/sharedb/blob/v1.1.0/lib/client/connection.js#L58-L61
https://github.com/share/sharedb/blob/v1.1.0/lib/backend.js#L112-L115

File: types/sharedb/index.d.ts (hunk 1 above)
Comment (eveshum) [reply types/sharedb/index.d.ts]: omg im going to remove myself...

**Review by ericyhwang — APPROVED** [review]

**Pull request merged**